hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/AvroSchemaEvolutionUtils.java

```diff
@@ -140,10 +140,14 @@ public static Schema reconcileSchema(Schema incomingSchema, Schema oldTableSchem
    * @return schema (based off {@code source} one) that has nullability constraints and datatypes reconciled
    */
   public static Schema reconcileSchemaRequirements(Schema sourceSchema, Schema targetSchema, Map<String, String> opts) {
-    if (sourceSchema.getType() == Schema.Type.NULL || sourceSchema.getFields().isEmpty() || targetSchema.getFields().isEmpty()) {
+    if (targetSchema.getType() == Schema.Type.NULL || targetSchema.getFields().isEmpty()) {
```

Comment:
This will also be problematic when transformation schema is not fully compatible with source schema. Can we take the approach if the input df is empty, do not set schema ?

Reply:
Balaji: I guess up until 0.14.0, we had isempty check explicitly for df that we consume from source before ingesting to hudi. but due to perf reasons, we removed that check. Hence we are running into this issue post 0.14.0 
So, would prefer to avoid checking df.isEmpty in this case.

Reply:
if its empty df i.e. 0 records, its ok to go ahead and make an empty commit w/ schema set to target schema. So, I don't see much necessity. Let me know wdyt